utils/manifest-utils/src/matching/mod.rs

```diff
@@ -0,0 +1,49 @@
+pub mod domain;
+
+use std::{collections::HashSet, fmt::Debug};
+
+use regex::RegexSet;
+
+use crate::ArgMatch;
+
+trait MatchPattern {
+    fn value(&self) -> String;
+    fn match_type(&self) -> ArgMatch;
+}
+
+pub trait Matcher: Debug + Send + Sync {
+    fn matches(&self, txt: &str) -> bool;
+}
```

Comment:
Please document what these traits do and their methods.

Reply:
The first one had even no implementation. Got removed. Weird that `clippy` had not reported it.
Second. Srsly. `Matcher` `matches` `txt`. Code cannot be more self descriptive than this.
I think I did not named it "Pattern", because I felt like it indicates some regex or glob, and matcher felt more general. Maybe I should.
Done first, thinking on second.

Reply:
Right, I was more concerned about `MatchPattern`, if it's removed and there's nothing useful to say about `Matcher`, then it's fine by me.